internal/app/go-filecoin/node/builder.go

```diff
@@ -124,6 +128,32 @@ func JournalConfigOption(jrl journal.Journal) BuilderOpt {
 	}
 }
 
+// MonkeyPatchNetworkParamsOption returns a function that sets global vars in the
+// binary's specs actor dependency to change network parameters that live there
+func MonkeyPatchNetworkParamsOption(params *config.NetworkParamsConfig) BuilderOpt {
+	return func(c *Builder) error {
+		if params.ConsensusMinerMinPower > 0 {
+			power.ConsensusMinerMinPower = big.NewIntUnsigned(params.ConsensusMinerMinPower)
+		}
+		if len(params.ReplaceProofTypes) > 0 {
+			miner.SupportedProofTypes = make(map[abi.RegisteredProof]struct{})
+			for _, proofType := range params.ReplaceProofTypes {
+				miner.SupportedProofTypes[abi.RegisteredProof(proofType)] = struct{}{}
+			}
+		}
+		return nil
+	}
+}
+
+// MonkeyPatchAddProofTypeOption returns a function that sets package variable
+// SuppurtedProofTypes to include the given registered proof type
+func MonkeyPatchAddProofTypeOption(proofType abi.RegisteredProof) BuilderOpt {
```

Comment:
I would replace rather than add here too, but AFAICT this is fine in the places it's used.

We'll likely get a chain validation test one day that asserts that an invalid proof type is rejected, where that invalid type is actually valid in testnet.